graphql_compiler/fast_introspection.py

```diff
@@ -0,0 +1,162 @@
+        "deprecationReason": __Field.fields["deprecationReason"].resolve(field, None),
+    }
+
+
+def _get_enum_value(enum_value: Tuple[str, GraphQLEnumValue]) -> Dict[str, Any]:
+    return {
+        "name": __EnumValue.fields["name"].resolve(enum_value, None),
+        "description": __EnumValue.fields["description"].resolve(enum_value, None),
+        "isDeprecated": __EnumValue.fields["isDeprecated"].resolve(enum_value, None),
+        "deprecationReason": __EnumValue.fields["deprecationReason"].resolve(enum_value, None),
+    }
+
+
+def _get_full_type(type_: GraphQLNamedType, schema: GraphQLSchema) -> Dict[str, Any]:
+    fields = __Type.fields["fields"].resolve(type_, None, includeDeprecated=True)
+    input_fields = __Type.fields["inputFields"].resolve(type_, None)
+    interfaces = __Type.fields["interfaces"].resolve(type_, None)
+    enum_values = __Type.fields["enumValues"].resolve(type_, None, includeDeprecated=True)
+
+    # The possibleTypes resolver requires GraphQLResolveInfo, so we just duplicate the resolver code
+    # here instead of having to mock GraphQLResolveInfo accurately. Moreover, GraphQLResolveInfo
+    # is for graphql-core internal use only, so it changes frequently.
+    possible_types = (
+        schema.get_possible_types(cast(GraphQLAbstractType, type_))
+        if is_abstract_type(type_)
+        else None
+    )
+
+    return {
+        "kind": __TypeKind.serialize(__Type.fields["kind"].resolve(type_, None)),
+        "name": __Type.fields["name"].resolve(type_, None),
+        "description": __Type.fields["description"].resolve(type_, None),
+        "fields": [_get_field(field) for field in fields] if fields else None,
+        "inputFields": (
+            [_get_input_value(field) for field in input_fields] if input_fields else None
+        ),
+        "interfaces": (
+            None if interfaces is None else [_get_type_ref(interface) for interface in interfaces]
+        ),
+        "enumValues": (
+            [_get_enum_value(enum_value) for enum_value in enum_values] if enum_values else None
+        ),
+        "possibleTypes": (
+            [_get_type_ref(possible_type) for possible_type in possible_types]
+            if possible_types
+            else None
+        ),
+    }
+
+
+def _get_directive(directive: GraphQLDirective) -> Dict[str, Any]:
+    return {
+        "name": __Directive.fields["name"].resolve(directive, None),
+        "description": __Directive.fields["description"].resolve(directive, None),
+        "locations": [
+            __DirectiveLocation.serialize(location)
+            for location in __Directive.fields["locations"].resolve(directive, None)
+        ],
+        "args": [
+            _get_input_value(arg) for arg in __Directive.fields["args"].resolve(directive, None)
+        ],
+    }
+
+
+def execute_fast_introspection_query(schema: GraphQLSchema) -> Dict[str, Any]:
+    """Compute the GraphQL introspection query."""
```

Comment:
This function is part of the user-facing public API of this module. As such, it needs quite a bit more detail in its docstring than this — otherwise, as a user, I probably wouldn't know what problem it solves for me, when it is appropriate to use it, and what functionality it replaces. That would mean it probably wouldn't get used, since users don't use features they don't understand or recognize the need for.

One especially important thing that we need to consider explicitly is this: how would a user know if _their_ introspection query is the same one that this module knows how to compute quickly? If the user is in-process and is therefore using `graphql-core` of the same version as the compiler, that's fine — but users don't have to do that. What if the user is using e.g. GraphiQL, which decides to send an introspection query to a server that is running this code? How would the server know if it can use this code to speed up the introspection or not? I don't think the current API solves this problem that users are very likely to run into, and I'd encourage you to revisit the end-to-end user experience and usage of this code in more detail.